File: spark-event-ui/src/app/model/sql/metrics/MetricDescription.ts
export type EnumMetricType = 'sum' | 'size' | 'timing' | 'nsTiming' | 'average';

export type EnumMetricUnit = 'bytes' | 'records' | 'partitions' | 'blocks' | 'files' | 'batches' |
  'count' | 'millis' | 'chunks' | 'unit?';

/**
 * Abstract class representing a metric type in Spark's event UI.
 */
export class MetricDescription {

  readonly name: string;
  readonly type: EnumMetricType;
  readonly unit: EnumMetricUnit;

  constructor(name: string, type: EnumMetricType, unit: EnumMetricUnit) {
    this.name = name;
    this.type = type;
    this.unit = unit;
  }

}



export class GenericMetricDescription extends MetricDescription {
  constructor(name: string, type: EnumMetricType) {
    super(name, type, 'unit?');
  }
}

File: spark-event-gateway-srv/src/main/java/fr/an/spark/gateway/sql/metrics/MetricDescription.java
package fr.an.spark.gateway.sql.metrics;

import lombok.Getter;

@Getter
public abstract class MetricDescription {

    public final String name;

    public enum EnumMetricType {
        SUM,
        SIZE,
        TIMING,
        NS_TIMING,
        AVERAGE
    }

    public final EnumMetricType type;

    public enum EnumMetricUnit {
        BYTES,
        RECORDS,
        PARTITIONS,
        BLOCKS,
        FILES,
        BATCHES,
        TASKS,
        COUNT,
        MILLIS,
        CHUNKS, UNKNOWN_UNIT
    }

    public final EnumMetricUnit unit;

    // -----------------------------------------------------------------------------------------------------------------

    public MetricDescription(String name, EnumMetricType type, EnumMetricUnit unit) {
        this.name = name;
        this.type = type;
        this.unit = unit;
    }

    // -----------------------------------------------------------------------------------------------------------------


    public static class GenericMetricDescription extends MetricDescription {
        public GenericMetricDescription(String name, EnumMetricType type) {
            super(name, type, EnumMetricUnit.UNKNOWN_UNIT);
        }
    }

}
